src/request-cards/Request-Card.test.js

```diff
@@ -0,0 +1,78 @@
+import { render, screen, act } from '@testing-library/react';
+import { RequestCards } from './RequestCards';
+import { data } from '../data.ts';
+import { BrowserRouter } from 'react-router-dom';
+import '@testing-library/jest-dom/extend-expect';
+import userEvent from '@testing-library/user-event';
+
+describe('Request Card Initial state', () => {
+  test('Request card has a title, description, category, upvote count, and comment count', () => {
+    render(
+      <BrowserRouter>
+        <RequestCards productRequests={data.productRequests} />
+      </BrowserRouter>,
+    );
+    screen.getAllByRole('heading', { level: 3 });
+    screen.getAllByTestId('description');
+    screen.getAllByTestId('category');
+    screen.getAllByTestId('upvotes');
+    screen.getAllByTestId('comments');
+  });
+});
+
+describe('Buttons functionality', () => {
+  test('Button should start with a numerical value and when clicked should increase by 1', async () => {
+    const user = userEvent.setup();
+    const baseProps = {
+      addUpVote: jest.fn(),
+      disabledUpvotes: [],
+    };
+    render(
+      <BrowserRouter>
+        <RequestCards {...baseProps} productRequests={data.productRequests} />
+      </BrowserRouter>,
+    );
+    const upvotesBtn = screen.getAllByTestId('upvotes');
+    expect(upvotesBtn[0]).toHaveTextContent('112');
+
+    await user.click(upvotesBtn[0]);
+    expect(baseProps.addUpVote).toHaveBeenCalledTimes(1);
+  });
+
+  test('Button should start as enabled', async () => {
+    const user = userEvent.setup();
+    const baseProps = {
+      addUpVote: jest.fn(),
+      disabledUpvotes: [],
+    };
+    render(
+      <BrowserRouter>
+        <RequestCards {...baseProps} productRequests={data.productRequests} />
+      </BrowserRouter>,
+    );
+
+    const upvotesBtn = screen.getAllByTestId('upvotes');
+    expect(upvotesBtn[0]).toHaveProperty('disabled', false);
```

Comment:
Is this right? The test says the button should start as true, but the disabled prop is false on this line

Reply:
Removed this test, will have to test it in App.